lib/train/transports/ssh_connection.rb

```diff
@@ -113,7 +114,7 @@ def wait_until_ready
         message: "Waiting for SSH service on #{@hostname}:#{@port}, " \
                  "retrying in #{delay} seconds",
       )
-      execute(PING_COMMAND.dup)
+      run_command(PING_COMMAND.dup)
```

Comment:
I think we want to use the `run_command_via_connection` here instead. The run_command has the potential for caching which could cause unexpected results.